test/brsTypes/components/Regex.test.js

```diff
@@ -0,0 +1,192 @@
+const brs = require("brs");
+const { BrsBoolean, BrsInvalid, BrsString, Int32, Regex, ValueKind } = brs.types;
+const { Interpreter } = require("../../../lib/interpreter");
+
+describe("Regex", () => {
+    let interpreter;
+
+    beforeEach(() => {
+        interpreter = new Interpreter;
+    });
+
+    describe("isMatch", () => {
+        it("matches strings with case sensitive flag", () => {
+            let regx = new Regex(new BrsString("hello_[0-9]*_world"));
+            let isMatch = regx.getMethod("isMatch");
+            expect(isMatch).toBeTruthy();
+
+            let str1 = isMatch.call(interpreter, new BrsString("hello_123_world"));
+            expect(str1.kind).toBe(ValueKind.Boolean);
+            expect(str1).toBe(BrsBoolean.True);
+
+            let str2 = isMatch.call(interpreter, new BrsString("HeLlO_123_WoRlD"));
+            expect(str2.kind).toBe(ValueKind.Boolean);
+            expect(str2).toBe(BrsBoolean.False);
+        });
+
+        it("matches strings with case insensitive flag", () => {
+            let regx = new Regex(new BrsString("hello_[0-9]*_world"), new BrsString("i"));
+            let isMatch = regx.getMethod("isMatch");
+            expect(isMatch).toBeTruthy();
+
+            let str1 = isMatch.call(interpreter, new BrsString("HeLlO_123_WoRlD"));
+            expect(str1.kind).toBe(ValueKind.Boolean);
+            expect(str1).toBe(BrsBoolean.True);
+
+            let str2 = isMatch.call(interpreter, new BrsString("HeLlO_XXX_WoRlD"));
+            expect(str2.kind).toBe(ValueKind.Boolean);
+            expect(str2).toBe(BrsBoolean.False);
+        });
+
+        it("matches a string in ISO8601 format", () => {
+            let regx = new Regex(new BrsString("P(\\d+Y)?(\\d+M)?(\\d+D)?T(\\d+H)?(\\d+M)?(\\d+(.\\d+)?S)?"));
```

Comment:
**woah** these are some crazy regexes.  Thanks for adding some real-world cases!

Reply:
😸